pkg/products/amqonline/reconciler.go

```diff
@@ -232,17 +242,127 @@ func (r *Reconciler) reconcileTemplates(ctx context.Context, serverClient k8scli
+	}
+
+	// Read the CRO secret, to get values to copy to enmasse namespace.
+	croSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.inst.Namespace,
+			Name:      postgresqlName,
+		},
+	}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: croSecret.Name, Namespace: croSecret.Namespace}, croSecret)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			return integreatlyv1alpha1.PhaseAwaitingComponents, nil
+		}
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	// Copy secret to enmasse namespace (adjust keys for keycloak)
+	keycloakSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Config.GetNamespace(),
+			Name:      postgresqlName,
+		},
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, keycloakSecret, func() error {
+		// I think it would be better to set the owner here to be the
+		// AuthService, but I can't immediately figure out how to get a
+		// reference to the scheme.
+		owner.AddIntegreatlyOwnerAnnotations(keycloakSecret, r.inst)
+
+		if keycloakSecret.Data == nil {
+			keycloakSecret.Data = make(map[string][]byte, 2)
 		}
+
+		keycloakSecret.Data["database-user"] = croSecret.Data["username"]
+		keycloakSecret.Data["database-password"] = croSecret.Data["password"]
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed reconciling enmasse standard auth service keycloak secret: %w", err)
 	}
+
+	standardAuthSvc := &enmasseadminv1beta1.AuthenticationService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "standard-authservice",
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, standardAuthSvc, func() error {
+		owner.AddIntegreatlyOwnerAnnotations(standardAuthSvc, r.inst)
+		standardAuthSvc.Spec.Type = "standard"
+
+		if standardAuthSvc.Spec.Standard == nil {
+			standardAuthSvc.Spec.Standard = &enmasseadminv1beta1.AuthenticationServiceSpecStandard{}
+		}
+		if standardAuthSvc.Spec.Standard.Datasource == nil {
+			standardAuthSvc.Spec.Standard.Datasource = &enmasseadminv1beta1.AuthenticationServiceSpecStandardDatasource{}
+		}
+
+		standardAuthSvc.Spec.Standard.Datasource.CredentialsSecret = corev1.SecretReference{
+			Name:      keycloakSecret.Name,
+			Namespace: keycloakSecret.Namespace,
+		}
+		standardAuthSvc.Spec.Standard.Datasource.Type = enmasseadminv1beta1.PostgresqlDatasource
+		standardAuthSvc.Spec.Standard.Datasource.Database = string(croSecret.Data["database"])
+		standardAuthSvc.Spec.Standard.Datasource.Host = string(croSecret.Data["host"])
+		standardAuthSvc.Spec.Standard.Datasource.Port, err = strconv.Atoi(string(croSecret.Data["port"]))
+		return err // either nil or failed to convert port to an int
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed reconciling standard AuthenticationService: %w", err)
+	}
+
+	// TODO: wait for auth service pod to be ready?
```

Comment:
Is this TODO required for this pr or is it a future wish ?

Reply:
Good catch! I'll remove that. According to @robshelly, in a future version, the phase in the AuthenticationService CR will be updated when everything is ready, so it will be easier to do that then.

Reply:
low hanging fruit :cherries: :banana: :apple:

Reply:
We probably don't need this tbh: I don't think anything in future stages depends on the auth services, so there's no need to block on their readiness. If they don't get to ready then an alert will fire and someone will have to deal with it :slightly_smiling_face: